MathlibTest/CasesLinter.lean

```diff
@@ -0,0 +1,13 @@
+import Mathlib.Tactic.Linter.CasesLinter
+
+set_option linter.cases true
+/--
+warning: The `cases'` tactic is discouraged: please strongly consider using `obtain`, `rcases` or `cases` instead.
+note: this linter can be disabled with `set_option linter.cases false`
+---
+warning: The `cases'` tactic is discouraged: please strongly consider using `obtain`, `rcases` or `cases` instead.
+note: this linter can be disabled with `set_option linter.cases false`
+-/
+#guard_msgs in
+example {a : (True ∨ True) ∨ (True ∨ True)}: True := by
+  cases' a with b b <;> cases' b <;> trivial
```

Comment:
I wonder if a test covering e.g. cases' in a sub-goal could be useful. Up to you.

Reply:
I think this is fine enough.